### PR TITLE
Affiliate SP/GP

### PR DESCRIPTION
**Pull request by lekshmimallika-aot**
*Issue #12303
https://github.com/bcgov/entity/issues/12303

*Description of changes:*
Added the party validation for firms (SP/GP)
Updated tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-api/src/auth_api/services/affiliation.py [hunk 1]
```diff
@@ -275,3 +283,23 @@ def _get_nr_details(nr_number: str, token: str):
...
+    def _validate_firms_party(token, business_identifier, party_name_str: str):
+        parties_url = current_app.config.get('LEGAL_API_URL') + f'/businesses/{business_identifier}/parties'
+        try:
+            lear_response = RestService.get(parties_url, token=token)
+        except (HTTPError, ServiceUnavailableException) as e:
+            current_app.logger.info(e)
+            raise BusinessException(Error.DATA_NOT_FOUND, None) from e
+        parties_json = lear_response.json()
+        for party in parties_json['parties']:
+            if party.get('officer').get('partyType') == 'organization':
```
Comment (seeker25) [inline auth-api/src/auth_api/services/affiliation.py:297]: I'd recommend using:
```
officer = party.get('officer')
```

**Comment by seeker25** [comment]
One more minor change:

image

**Comment by pwei1018** [comment]
> One more minor change (I think), this should provide the ENV variables in DEV/TEST/PROD:
> 
> image
> 
> Similar to this:
> 
> https://github.com/bcgov/lear/blob/e259eb1fb42c77b8e43b020081ef25558ff5ceea/legal-api/devops/vaults.json#L21

image

https://github.com/bcgov/business-edit-ui/blob/main/devops/vaults.json

File: auth-api/src/auth_api/config.py [hunk 2]
```diff
@@ -111,6 +111,8 @@ class _Config:  # pylint: disable=too-few-public-methods
     NOTIFY_API_URL = os.getenv('NOTIFY_API_URL')
     BCOL_API_URL = os.getenv('BCOL_API_URL')
     PAY_API_SANDBOX_URL = os.getenv('PAY_API_SANDBOX_URL')
+    LEGAL_API_VERSION = os.getenv('LEGAL_API_VERSION')
```
Comment (pwei1018) [inline auth-api/src/auth_api/config.py:114]: Please add 'legal-api' into API vault in vaults.json

**Review by pwei1018 — DISMISSED** [review]

**Pull request merged**